### PR TITLE
More flexible InputVariables layout and changes to Lineplot visualization

### LINKED ISSUE #1012: Improve UI for 3 main sections requiring user input on Line plots
Discussed in 4/4/2022 EDA UX meeting

There are 3 main sections requiring user input on line plots.

1. X-axis and Y-axis variable selectors → standard on all visualization tools
2. Y-axis aggregation function selector → specific to line plots
3. Stratification variable selectors → standard on all visualization tools

Improvements needed:

- [x] The UX for these 3 sections needs better organization, perhaps there can be 3 columns for these 3 sections, instead of having the Y-axis aggregation selector below the other two. Place the y-axis aggregation function selector section between the other 2 sections, as it goes along with the y-axis variable selector and is required for making the plots.

- [x] Gray out the aggregation function selector until the y-axis variable is chosen (currently, the function can be selected 1st, and no constraints are imposed on the variables)

- [x] Remove the title from the numerator / denominator selector that appears when proportions are chosen to aggregate the data

- [x] For proportion mode, place a division line or symbol between the numerator and denominator selectors to make what the selections are doing more clear

### PR DESCRIPTION
**Pull request by bobular**
Resolves #1012

I refactored away some of the repetition in `InputVariables`

Then, by adding this to the props 

```
	  customSections={[
	    {
	      title: "Crazy new section",
	      order: 75,
	      content: <span>just a span just a span just a span just a span just a span just a span just a span just a span </span>
	    }
	  ]}
```

you can get a new "section" in whichever position you like (via the `order` prop)

image

Going to use this for the requested changes to Lineplot

### CODE REVIEW

**Comment by bobular** [comment]
I'm not really sure why the "Axis variables" `.inputGroup` takes so much space, and adding `justifyContent: 'space-between'` to the `.inputs` container doesn't do anything!

image

**Comment by bobular** [comment]
Also, the horizontal line (done with an old school `<hr />`) doesn't look so nice on Chrome (above screenshots are FF)

(I've zoomed it a bit to make it clear that there is some shadow/3D stuff going on in Chrome)
image

**Comment by bobular** [comment]
The web-components upgrade is for https://github.com/VEuPathDB/web-components/pull/327 (I merged one of Connor's PRs so I could work on `InputVariables` more easily, but didn't know he was going to upgrade components before merging)

**Comment by bobular** [comment]
I guess it would be good to stop the radio buttons wrapping, e.g. here
image

File: src/lib/core/components/visualizations/InputVariables.tsx [hunk 1]
```diff
@@ -20,13 +20,45 @@ interface InputSpec {
   role?: 'axis' | 'stratification';
 }
 
+interface SectionSpec {
+  order: number;
+  title: string | ReactNode;
```
Comment (jtlong3rd) [inline src/lib/core/components/visualizations/InputVariables.tsx:25]:
```suggestion
  title: ReactNode;
```

(Strings are `ReactNode`s.)

**Review by jtlong3rd — APPROVED** [review]

**Pull request merged**